Add user management components and functionality

Mustfa-IT opened the pull request:
Introduce UserProfile and UserTable components for managing users, along with modal support for infractions and enhanced search functionality. Implement a cancellable fetch for user data and update the user model to include additional fields. Add InfractionsTab and EconomyTab components for displaying user infractions and economic data, respectively. Enhance DiscordUserTable with avatar support and improved user status display.

Mustafa-IT-cave committed:
Add UserProfile and UserTable components for user management; implement modal for infractions and search functionality

Mustafa-IT-cave committed:
Add user model types and implement cancellable fetch for user data; enhance UserTable and DiscordUserTable components with search functionality and loading states

Mustafa-IT-cave committed:
Add InfractionsTab and EconomyTab components; update User model to include discord_img and enhance DiscordUserTable with avatar support

BogTheMudWing commented:
![image](https://github.com/user-attachments/assets/f6de25ee-3e25-4334-aef6-fe757086d2cb)

Warn Channel ID should not be specified here. Leave it null and DreamvisitorHub should create a channel set this value automatically (if "Send a warn message" is checked)

Mustfa-IT commented:
OK

Commit:
Revise formatting and presentation

BogTheMudWing commented:
And description of item should be plain text, not rich text. Any formatting should be done with markdown syntax.
That is a more of a DB thing but I thought I should mention it

Commit:
Remove warn channel and add spacing

The warn channel ID does not need to be displayed. Space is added between the expired pill and the dropdown arrow in the infractions list.

Mustafa-IT-cave committed:
Update UserProfile to include send_warning option for infractions; modify related types and form handling

Mustafa-IT-cave committed:
Merge branch 'feature-users-page' of https://github.com/WOFTNW/dreamvisitor-web into feature-users-page

BogTheMudWing commented:
![image](https://github.com/user-attachments/assets/da0120ba-e676-453f-9b4a-1afb18aa3289)

I removed the warn channel from the infractions list and added some space between the expired pill (active here) and the dropdown arrow.

Mustfa-IT commented:
i added the send warning to the model and made it work with DB

Mustfa-IT commented:
> And description of item should be plain text, not rich text. Any formatting should be done with markdown syntax. That is a more of a DB thing but I thought I should mention it

it is plain text i may have forgotten to update the page or something , butin DB it is plain Text
